### PR TITLE
Add NumPy 2 support

### LINKED ISSUE #493: OverflowError: Python integer 256 out of bounds for uint8
I am breaking the demos in the notebook in individual to get acquainted with the package to learn how it works and adapt it to my needs.

Running demo4 as a separate script with python I get this error traceback:

```
D:\Users\xxx\Work\__WFDB\wfdb-python>python demo4.py
Traceback (most recent call last):
  File "D:\Users\xxx\Work\__WFDB\wfdb-python\demo4.py", line 12, in <module>
    annotation = wfdb.rdann('sample-data/100', 'atr', sampfrom=100000, sampto=110000)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Users\xxx\Work\__WFDB\wfdb-python\wfdb\io\annotation.py", line 1953, in rdann
    (sample, label_store, subtype, chan, num, aux_note) = proc_ann_bytes(
                                                          ^^^^^^^^^^^^^^^
  File "D:\Users\xxx\Work\__WFDB\wfdb-python\wfdb\io\annotation.py", line 2154, in proc_ann_bytes
    sample_diff, current_label_store, bpi = proc_core_fields(filebytes, bpi)
                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Users\xxx\Work\__WFDB\wfdb-python\wfdb\io\annotation.py", line 2240, in proc_core_fields
    sample_diff += int(filebytes[bpi, 0] + 256 * (filebytes[bpi, 1] & 3))
                                           ~~~~^~~~~~~~~~~~~~~~~~~~~~~~~
OverflowError: Python integer 256 out of bounds for uint8
```

The same happens with Linux (Ubuntu 22.04).

After changing the highlighted line it works like the notebook.
```
$ git diff
diff --git a/wfdb/io/annotation.py b/wfdb/io/annotation.py
index b398fa0..4d451b1 100644
--- a/wfdb/io/annotation.py
+++ b/wfdb/io/annotation.py
@@ -2237,7 +2237,7 @@ def proc_core_fields(filebytes, bpi):

     # Not a skip - it is the actual sample number + annotation type store value
     label_store = filebytes[bpi, 1] >> 2
-    sample_diff += int(filebytes[bpi, 0] + 256 * (filebytes[bpi, 1] & 3))
+    sample_diff += int(filebytes[bpi, 0]) + 256 * int(filebytes[bpi, 1] & 3)
     bpi = bpi + 1

     return sample_diff, label_store, bpi
```

### PR DESCRIPTION
**Pull request by cbrnr**
Fixes #493.

### CODE REVIEW

**Comment by cbrnr** [comment]
@tompollard you opened #506, #507, and #508 – do the changes in this PR not solve the issue? All tests pass with NumPy 2.

File: pyproject.toml [hunk 1]
```diff
@@ -11,7 +11,7 @@ license = "MIT"
 
 [tool.poetry.dependencies]
 python = ">=3.7"
-numpy = ">=1.10.1,<2.0.0"
+numpy = ">=2.0.0"
```
Comment (cbrnr) [inline pyproject.toml:14]: This could even remain `numpy = ">=1.26.0"`, since all changes should work with both NumPy 1.x and NumPy 2.x.

**Comment by tompollard** [comment]
@cbrnr thanks for looking at this and sorry for the duplicate effort.

> @tompollard you opened https://github.com/MIT-LCP/wfdb-python/pull/506, https://github.com/MIT-LCP/wfdb-python/pull/507, and https://github.com/MIT-LCP/wfdb-python/pull/508 – do the changes in this PR not solve the issue? All tests pass with NumPy 2.

I think the change in this pull request resolves one of the issues, but not all of them. Errors may not be obvious because (1) the test suite doesn't have full coverage and (2) overflows don't raise errors by default.

I have tried to address the warnings raised when running tests with `NPY_PROMOTION_STATE=weak_and_warn`, but I think more work is needed (e.g. by extending tests).

We may want to set the package to raise error on overflow when in development, which I think we can do with `np.seterr` (https://numpy.org/doc/stable/reference/generated/numpy.seterr.html)

**Comment by cbrnr** [comment]
No problem! Feel free to take whatever you need from this PR, and let me know if you need further help. It would be great if NumPy 2 support wasn't too far in the future!

**Pull request closed without merging**